sql/hive-thriftserver/src/main/java/org/apache/hive/service/cli/session/HiveSessionImpl.java

```diff
@@ -141,7 +141,7 @@ public void open(Map<String, String> sessionConfMap) throws HiveSQLException {
     sessionState = new SessionState(hiveConf, username);
     sessionState.setUserIpAddress(ipAddress);
     sessionState.setIsHiveServerQuery(true);
-    SessionState.start(sessionState);
+    SessionState.setCurrentSessionState(sessionState);
```

Comment:
hmm just curious why we don't need to start the new session in these places. Maybe add more details in the PR description?

Reply:
Spark SQL does not need SessionDirs.

Reply:
Is it safe when we use `ADD JARS`?

Reply:
And, I'm wondering about what happens to `LOCALSCRATCHDIR` after this PR?

Reply:
Yes. It is safe when use `ADD JARS`. We have disabled creating these directories for more than a year with the following changes(`HiveConf.ConfVars.WITHSCRATCHDIR=false`):

![image](https://user-images.githubusercontent.com/5399861/116785447-312cc500-aacc-11eb-8dff-6ae75fbbc4d7.png)